base/uk.ac.stfc.isis.ibex.ui/src/uk/ac/stfc/isis/ibex/ui/tables/ColumnComparator.java

```diff
@@ -0,0 +1,68 @@
+package uk.ac.stfc.isis.ibex.ui.tables;
+
+
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.jface.viewers.ViewerComparator;
+import org.eclipse.swt.SWT;
+
+/**
+ * Compares columns in a table to be sorted.
+ * 
+ * This class sorts based on the string representation of the cell. To sort based on something else, subclass this and override the method compare.
+ * 
+ * @param <T> The type of object stored in the table.
+ */
+public class ColumnComparator<T> extends ViewerComparator {
+	protected int propertyIndex;
+    protected int direction = 0;
+    protected static final int DESCENDING = 1;
```

Comment:
Would this be cleaner as a boolean rather than an int? `protected boolean sortAscending = True`